modules/local/report_aggregate.nf

```diff
@@ -12,12 +12,16 @@ process REPORT_SUMMARIES{
 
     output:
     path("final_report.tsv"), emit: final_report
-    path("*_flattened.json"), emit: flattened_files
+    path("final_report_transposed.tsv"), emit: final_report_transposed
+    path("final_report_flattened.json"), emit: flattened_files
+    path("*${sample_flat_suffix}"), emit: flat_samples
+    val sample_flat_suffix, emit: sample_suffix
     path "versions.yml", emit: versions
 
     script:
+    sample_flat_suffix = "_flat_sample.json"
     """
-    report_summaries.py -f ${summary_report} -o final_report.tsv
+    report_summaries.py -f ${summary_report} -o final_report.tsv -s ${sample_flat_suffix}
```

Comment:
So the while compressing the files in the Python script would be the most efficient approach, I opted against doing so as I needed to *create an output channel with the formatted meta information for the nf-iridanext plug-in* resulting in the need to create a process to construct an output channel containing the information.


I decided having a generic GZIP_FILE process would atleast offer additional utility as it could be used in other places in the workflow.

Reply:
As-discussed we will re-visit this later when we have time.